web/modules/custom/dpl_event/src/EventSubscriber/EventsRedirect.php

```diff
@@ -0,0 +1,108 @@
+<?php
+
+namespace Drupal\dpl_event\EventSubscriber;
+
+use Drupal\Core\Entity\EntityTypeManagerInterface;
+use Drupal\recurring_events\Entity\EventInstance;
+use Drupal\recurring_events\Entity\EventSeries;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\HttpFoundation\RedirectResponse;
+use Symfony\Component\HttpKernel\Event\RequestEvent;
+use Symfony\Component\HttpKernel\KernelEvents;
+
+/**
+ * Redirects requests for eventseries entities.
+ *
+ * @package Drupal\dpl_event\EventSubscriber
+ */
+class EventsRedirect implements EventSubscriberInterface {
+
+  /**
+   * The entity type manager.
+   *
+   * @var \Drupal\Core\Entity\EntityTypeManagerInterface
+   */
+  protected $entityTypeManager;
+
+  /**
+   * Constructs a new EventsRedirect object.
+   *
+   * @param \Drupal\Core\Entity\EntityTypeManagerInterface $entity_type_manager
+   *   The entity type manager.
+   */
+  public function __construct(EntityTypeManagerInterface $entity_type_manager) {
+    $this->entityTypeManager = $entity_type_manager;
```

Comment:
Consider changing this to constructor property promotion like we have for `EventSeriesTabAccessCheck`.

You reduce the need for an explit property declaration and we also get actual type safety by setting the type on the propety in PHP instead of in a DocBlock.